zio-http/shared/src/main/scala/zio/http/Route.scala

```diff
@@ -131,6 +133,57 @@ sealed trait Route[-Env, +Err] { self =>
         Handled(rpm.routePattern, handler2, location)
     }
 
+  /**
+   * Effectfully peeks at the unhandled failure of this Route.
+   */
+  final def tapErrorZIO[Err1 >: Err](
+    f: Err => ZIO[Any, Err1, Any],
+  )(implicit trace: Trace, ev: CheckResponse[Err]): Route[Env, Err1] =
+    self match {
+      case Provided(route, env)                        => Provided(route.tapErrorZIO(f), env)
+      case Augmented(route, aspect)                    => Augmented(route.tapErrorZIO(f), aspect)
+      case Handled(routePattern, handler, location)    =>
+        Handled(
+          routePattern,
+          if (ev.isResponse) {
```

Comment:
This is always true. `Handled` has a fixed error type of `Nothing` and will pick up the `implicit val response: CheckResponse[Response]`

Handled Routes are by definition routes without error. This should just returned the route without any modification.
The becomes clear when substituting the types. Because f is `Nothing => ZIO[Any, Err1, Any]`